src/graph/topo.cc

```diff
@@ -667,6 +666,25 @@ ncclResult_t scclGetAlgoFromXMLAndSetComm(struct ncclComm* comm) {
   NCCLCHECK(xmlGetAttrInt(topNode, "nchunksperloop", &nchunksPerLoop));
   int globalNChannels;
   NCCLCHECK(xmlGetAttrInt(topNode, "nchannels", &globalNChannels));
+
+  const char* redop;
+  NCCLCHECK(xmlGetAttrStr(topNode, "redop", &redop));
+  if (strcmp(redop, "sum") == 0){
```

Comment:
why is the FUNC not enough? in what kinda of scenario we might need multiple?

Reply:
All ncclKernels and ncclFuncs take a FUNC argument, even those that do not use FUNC like AllGather.
By default in NCCL these kernels take Sum as the FUNC argument value. 
Similarly, CustomCollective kernels have SUM, Prod, and other as reduction operators.
This operator can be passed through ncclCustomCollective or through XML. I think passing it through XML is cleaner.